### ISSUE TITLE
Unserialize error

### ISSUE THREAD

**Issue by gnat42** [Block8/PHPCI#604]
Hello,

  I had evaluated phpci in a vm awhile back. Now we have an instance in production however the first project I add to it results in the following.

Notice: unserialize(): Error at offset 0 of 2 bytes in /path/to/phpci/PHPCI/Model/Project.php line 63

  This is a completely fresh install so I'm at a loss.

**Comment by subdee** [comment]
This is still an issue because I get the same on a fresh installation when using Gitlab repositories.

**Comment by subdee** [comment]
Update: It seems the problem occurs only if you install by downloading manually. Installation via composer doesn't have this problem.

**Comment by gnat42** [comment]
@subdee  you mean if you install phpci via composer this bug doesn't exist? But if you download a tarball it does have the bug?

**Comment by subdee** [comment]
Yes, first I followed the instructions given [https://www.phptesting.org/wiki/Installing-PHPCI](here) and I had this issue. Then I followed the instructions given [https://www.phptesting.org/install-phpci](here) and it worked fine. 

The instructions should actually be consistent, but that's a different issue.

**Comment by briedis** [comment]
I downloaded phpci archive manually, ran composer install and the bug persists.

**Comment by Baachi** [comment]
The issue persist, i also follow the instructions and ran into this issue

**Comment by Baachi** [comment]
Strange, i recreated the project and the issue doesn't persist anymore

**Comment by juliandortmund** [comment]
This problem occurs even in current versions.
I try to connect my GitLab server with 1.6.5 and the current 1.7.0-beta2
The protocol is https and the address ends with .git 
All I get is 
```bash
Sorry, there was a problem
Notice: unserialize(): Error at offset 0 of 2 bytes in /var/www/virtual/username/phpci/PHPCI/Model/Project.php line 100
```

**Comment by WitTrue** [comment]
I think the problem could be here: 

    protected function processAccessInformation(Project &$project)
    {
        $matches = array();
        $reference = $project->getReference();

        if ($project->getType() == 'gitlab') {
            $info = array();

            if (preg_match('`^(.+)@(.+):([0-9]*)\/?(.+)\.git`', $reference, $matches)) {
                $info['user'] = $matches[1];
                $info['domain'] = $matches[2];
                $info['port'] = $matches[3];

                $project->setReference($matches[4]);
            }

            $project->setAccessInformation($info);
        }
    }

A viable solution from end-user point of view is that you should not use the https url to the repository, but the ssh one.
So this: git@gitlab.com:user/project.git
Instead of this: https://gitlab.com/user/project.git

This because if you use the last one (https://gitlab.com/user/project.git), it will assign an empty array to accessInformation, that will create a problem in the following function (since $info will be = '[]', returning true to `substr($info, 0, 1) != '{'`): 

    public function getAccessInformation($key = null)
    {
        $info = $this->data['access_information'];

        // Handle old-format (serialized) access information first:

        if (!empty($info) && substr($info, 0, 1) != '{') {
            $data = unserialize($info);
        } else {
            $data = json_decode($info, true);
        }

        if (is_null($key)) {
            $rtn = $data;
        } elseif (isset($data[$key])) {
            $rtn = $data[$key];
        } else {
            $rtn = null;
        }

        return $rtn;
    }

I hope this will help.
